### PR TITLE
Adding reset and pause/resume functionality to joy

### PR DESCRIPTION
**Pull request by skyler237**
I added functions to reset the mav to the origin using the start button (xbox button 9) and to pause/resume the simulation using the back button (xbox button 8).

### CODE REVIEW

File: src/joy.cpp [hunk 1]
```diff
@@ -20,6 +20,11 @@
 
 
 #include "fcu_common/joy.h"
+#include "geometry_msgs/Pose.h"
+#include "geometry_msgs/Twist.h"
```
Comment (superjax) [inline src/joy.cpp]: I think that these are included already by gazebo_msgs/ModelState.h  If they are, there is no reason to include them again.

File: src/joy.cpp [hunk 2]
```diff
@@ -85,13 +97,70 @@ void Joy::StopMav() {
...
 
+/* Resets the mav back to origin */
+void Joy::ResetMav() 
+{
+	ROS_INFO("Mav position reset.");
+	ros::NodeHandle n;
+    geometry_msgs::Pose start_pose;
+	start_pose.position.x = 0.0;
+	start_pose.position.y = 0.0;
+	start_pose.position.z = 0.1;
```
Comment (superjax) [inline src/joy.cpp]: We might want to make these a parameter.  There are a few instances where we don't initialize the MAV at (0,0,0).  Definitely the defaults should be (0,0,0), but let's make it a parameter where it is reset to.

File: src/joy.cpp [hunk 3]
```diff
@@ -85,13 +97,70 @@ void Joy::StopMav() {
...
+
+	geometry_msgs::Twist start_twist;
+	start_twist.linear.x = 0.0;
+	start_twist.linear.y = 0.0;
+	start_twist.linear.z = 0.0;
+	start_twist.angular.x = 0.0;
+	start_twist.angular.y = 0.0;
+	start_twist.angular.z = 0.0;
+
+    gazebo_msgs::ModelState modelstate;
```
Comment (superjax) [inline src/joy.cpp]: Just some formatting stuff.  Weird indentations

File: src/joy.cpp [hunk 4]
```diff
@@ -85,13 +97,70 @@ void Joy::StopMav() {
...
+	modelstate.model_name = (std::string) mav_name_;
+	modelstate.reference_frame = (std::string) "world";
+	modelstate.pose = start_pose;
+	modelstate.twist = start_twist;
+
+    ros::ServiceClient client = n.serviceClient<gazebo_msgs::SetModelState>("/gazebo/set_model_state");
+    gazebo_msgs::SetModelState setmodelstate;
```
Comment (superjax) [inline src/joy.cpp]: Again, the weird indentations (we use 2 spaces)

File: src/joy.cpp [hunk 5]
```diff
@@ -85,13 +97,70 @@ void Joy::StopMav() {
...
+	setmodelstate.request.model_state = modelstate;
+	client.call(setmodelstate);
+}
+
+// Pauses the gazebo physics and time
+void Joy::PauseSimulation()
```
Comment (superjax) [inline src/joy.cpp:130]: Nice!  I like the pause and resume buttons.  That's awesome

**Comment by superjax** [comment]
All in all, these look good!  I have a few comments to clean up the code, but when that's ready, I'll Merge it in!  Great work.

**Comment by skyler237** [comment]
Here are the changes you requested!

**Pull request merged**